Mlem/App/Views/Pages/Editors/FilterViolationWarning.swift

```diff
@@ -0,0 +1,38 @@
+//
+//  FilterViolationWarning.swift
+//  Mlem
+//
+//  Created by Eric Andrews on 2025-02-03.
+//
+
+import SwiftUI
+
+struct FilterViolationWarning: View {
+    @Environment(Palette.self) var palette
+    
+    let failingText: String
+    let isPost: Bool
+    
+    var context: LocalizedStringResource {
+        isPost ? "Posts containing " : "Comments containing "
+    }
+    
+    var body: some View {
+        VStack(alignment: .leading, spacing: Constants.main.standardSpacing) {
+            (Text(Image(systemName: Icons.warning)) + Text(" Filter violation"))
+                .foregroundStyle(palette.warning)
+                .padding(Constants.main.standardSpacing)
+                .padding(.horizontal, Constants.main.halfSpacing)
```

Comment:
IMO we should make this badge a little smaller

```suggestion
                .font(.footnote)
                .foregroundStyle(palette.warning)
                .padding(.vertical, 5)
                .padding(.horizontal, 7)
```